src/Microsoft.Restier.Providers.EntityFramework/Model/ModelProducer.cs

```diff
@@ -62,7 +62,19 @@ public Task<IEdmModel> GetModelAsync(ModelContext context, CancellationToken can
             var dbContext = context.GetApiService<DbContext>();
 
             var efModel = (dbContext as IObjectContextAdapter).ObjectContext.MetadataWorkspace;
-            var efEntityContainer = efModel.GetItems<EntityContainer>(DataSpace.CSpace).Single();
+            var efEntityContainers = efModel.GetItems<EntityContainer>(DataSpace.CSpace);
+            var efEntityContainer = efEntityContainers.FirstOrDefault(c => c.Name == dbContext.GetType().Name);
+            if (efEntityContainer == null)
+            {
+                if (efEntityContainers.Count > 1)
+                {
```

Comment:
I believe that this code would work even if an individual EDMX file ever had multiple containers (though I don't know if that is actually even possible). Given the fact that right now, the code is pulling containers from totally separate `DbContext` instances and putting them in the same collection, one would think that multiple Containers in the same Context would also apply. However, if at some point in the future it didn't, the logic for `efEntityContainers` could be adjusted easily; and that's why I broke that call out from the `.FirstOrDefault()` call.